pkg/cmd/server/api/types.go

```diff
@@ -112,12 +112,30 @@ type NodeConfig struct {
 	// MasterClientConnectionOverrides provides overrides to the client connection used to connect to the master.
 	MasterClientConnectionOverrides *ClientConnectionOverrides
 
-	// DNSDomain holds the domain suffix
+	// DNSDomain holds the domain suffix that will be used for the DNS search path inside each container. Defaults to
+	// 'cluster.local'.
 	DNSDomain string
 
-	// DNSIP holds the IP
+	// DNSIP is the IP address that the DNS server will listen on and which will be reported into pods. Defaults
+	// to the service IP of the Kubernetes master. This IP must be listening on port 53 for compatibility with
+	// libc resolvers (which cannot be configured to resolve names from any other port). When running more complex
+	// local DNS configurations, this is often set to the local address of a DNS proxy like dnsmasq, which then
+	// will consult either the local DNS (see dnsBindAddress) or the master DNS to offer better availability.
 	DNSIP string
 
+	// DNSBindAddress is the ip:port to serve DNS on. If this is not set, the DNS server will not be started.
+	// Because most DNS resolvers will only listen on port 53, if you select an alternative port you will need
+	// a DNS proxy like dnsmasq to answer queries for containers. A common configuration is dnsmasq configured
```

Comment:
Because we say that in the previous field I don't know that it is necessary here (reordered to make that clear).